openeogeotrellis/deploy/batch_job.py

```diff
@@ -211,6 +271,208 @@ def epsg_code(gps_crs) -> Optional[int]:
 
     logger.info("wrote metadata to %s" % metadata_file)
 
+
+def read_projection_extension_metadata(asset_path: str) -> dict:
```

Comment:
Return type should be an `Optional[dict]`.

Reply:
Indeed.

In fact I dug a bit deeper with mypy and mypy found some more potential issues, so I decided to make the return and argument types more consistent overal.

I got rid of returning None because we can just as well return an empty dict in these cases. The check is the same but we don't do anything special with None. 
That avoids extra checks for None inside the functions, and it avoids that function arguments also need to have type Optional[dict] .

Further, I introduced two type aliases because there were a lot of Dict[str, Any] everywhere and that made it confusing what was the difference between dicts at the input (coming from gdal.Info) versus dicts at the output, namely the projection metadata in STAC format.